doc/developer_resources.rst

```diff
@@ -11,6 +11,18 @@ If CmdStan changes default values for arguments, pipe the output of ``modelbinar
 file and then point ``scripts/parse_cmdstan_help.py`` at that file. The output
 should replace the file ``httpstan/services/cmdstan-help-all.json``.
 
+DEBUG mode
+==========
+
+If the environment variable ``HTTPSTAN_DEBUG`` is set to ``1`` or ``true``, the
+call to the stan::services sampling function will block until finished instead
+of being run in the background. This should make debugging crashes with ``gdb``
+much easier.  There is, however, one complication. The number of samples drawn
+must be set to be a very low number (e.g., ``10``) otherwise the call to the
```

Comment:
This is fine

Is there a way to manually increase the socket size or automatically empty it?

Reply:
I did give this some thought. I don't think there's an easy way.

We can print a warning about the requirement that num_samples be set to 10 though (when HTTPSTAN_DEBUG is set). That should help a great deal.